### DISCUSSION TITLE
Is "Writing logs to Elasticsearch" actually about writing logs?

### DISCUSSION THREAD

**Discussion by makrushin-evgenii** [apache/airflow#31243] [Q&A]
Hi. I configured remote logging following the instruction ["Writing logs to Amazon S3"](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/logging/s3-task-handler.html). Works as expected: logs are written to S3 bucket, logs are read from this bucket for viewing in Airflow WebUI

I want the same thing, but with ElasticSearch. So that logs can be viewed not only in Airflow WebUI, but also in Kibana. I found a similar instruction ["Writing logs to Elasticsearch"](https://airflow.apache.org/docs/apache-airflow-providers-elasticsearch/stable/logging/index.html#writing-logs-to-elasticsearch) and they even affect one section of the config. Then I expect that they will work the same way. But I was confused by the line:
> Airflow can be configured to **read** task logs from Elasticsearch and **optionally write logs to stdout** in standard or json format.

Also, I didn't find write methods in [`es_task_handler.py`](https://github.com/apache/airflow/blob/main/airflow/providers/elasticsearch/log/es_task_handler.py) class

So. Will this instruction will help me to configure remote logging with ElasticSearch? If not, why is this provider configured in the `remote_logging` section?

How should remote logging work with Elastic? Where I can define ElasticSearch index to write and read logs, how will the logs get into Elastic?

**Comment by makrushin-evgenii** [comment]
The logs are supposed to be uploaded to ElasticSearch using fluentd and `es_task_handler` will only read them?

I found a report with the following scheme: [A look under the hood of the Airflow logging subsystem (17:28)](https://www.youtube.com/watch?v=5y55hBr2eXY&t=1048s)

**Comment by potiuk** [comment]
Try to configure it and check. 

All the task handlers that are implemented in Airflow write logs from tasks when configured (they are just airflow-specific Python loggers) following https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/logging-tasks.html - and all the logs produced by standard python loggers in tasks **should** be written by them. They also serve the purpose to be able to read the logs from target system, so that they can be shown in Airflow UI (note that we are only talking about task logs - if you have other component's logs  (scheduler, generic non-task worker logs, webserver, triggerer, etc. ) to be written to elasticsearch you need to configure it your deployment in the way you find best for your deployment - and there fluentd is possibly one of the options).

The task log handlers only handle writing and reading logs  for running tasks.

I have 0 experience with doing it -  so can't help with specific problems you might have and parameters to set - you need to figure it yourself, but following the docs you have and looking at the source code if needed, might be helpful.

And if  you manage to do it and find misleading things that led you to misunderstanding, A PR to fix those things is most welcome. Airflow (including docs) is created by almost 2500 to date, and they are mostly contributing things in their free time, so thre might be some mistakes or lack of information, quite often then people who find and solve the problem contribute back improvements and documentation fixes to help others and give back for the free software they got. IT's super-easy just click "Suggest a change on this page" and Pull Request will be open where doc updates can be added even without leaving GitHub UI.
